### DISCUSSION TITLE
Fix for Python error: `This environment is externally managed`

### DISCUSSION THREAD

**Discussion by havardthom** [community-scripts/ProxmoxVE#555] [Guides]
Original thread: https://github.com/tteck/Proxmox/discussions/3300

## TLDR: Execute `rm -rf /usr/lib/python3.*/EXTERNALLY-MANAGED` in the LXC console.

Users that manually updates an LXC to Debian 12.6 will reintroduce the `/usr/lib/python3.*/EXTERNALLY-MANAGED` file, causing the error below when installing python packages with pip. Remove this file by executing `rm -rf /usr/lib/python3.*/EXTERNALLY-MANAGED` in the LXC console.

If you install new scripts or update LXCs using the [LXC updater script](https://community-scripts.github.io/ProxmoxVE/scripts?id=update-lxcs) you will not have this problem.

```
× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.11/README.venv for more information.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
```

**Comment by MijnSpam** [comment]
Maybe good to know for others.
When trying to update an LXC and you get an error like: (example of Sabdnzb) 
"[ERROR] in line 66: exit code 0: while executing command python3 -m pip install -r requirements.txt &> /dev/null" << old helper scripts
Or with the new one scripts on line 43, this seems to be the fix as well.
Thought it was a update script error as it appeared to be this, with the TLDR fix made it working again.

**Reply by dominikwirtz** [reply]
With v2.14.7 Update of Paperless-ngx the rm -rf /usr/lib/python3.*/EXTERNALLY-MANAGED has no effect update still fails at:

[ERROR] in line 68: exit code 0: while executing command /usr/bin/python3 manage.py migrate &> /dev/null

Someone a new idea?

**Reply by michelroegl-brunner** [reply]
please run
 ` /usr/bin/python3 manage.py migrate ` in /opt/paperless/src to see the full error

**Reply by dominikwirtz** [reply]
RRORS:
?: (corsheaders.E014) Origin 'https://paperless.xxx.de,http://10.111.111.202:8000' in CORS_ALLOWED_ORIGINS should not have path

**Reply by michelroegl-brunner** [reply]
Is a error in your config then? U know the ip/domain?

**Reply by dominikwirtz** [reply]
yes after removing the IP update works again, thank you very much!
i have set the ip with Paperless URL now i added it to allowed hosts. and cors hosts. now its ok

**Reply by michelroegl-brunner** [reply]
Your welcome :)

**Comment by LaurenceBarnes** [comment]
https://stackoverflow.com/a/75722775

Wouldn't that be the better solution?
